### PR TITLE
Added get_ip_list() utility function

### PR DESCRIPTION
**Pull request by Script-Nomad**
Added a function for something I do pretty regularly, which is converting an IPv4 network into a list of their individual IP addresses as `str()`. The ipaddress library has a way of doing this via the IPv4Network.hosts() generator, but I find this is a nice convenience method. 

You can write it as `get_ip_list('192.168.1.0/24')`, `get_ip_list('192.168.1.0/255.255.255.0')`, `get_ip_list('192.168.1.0', mask='255.255.255.0'), or `get_ip_list('192.168.1.0', mask='24')

Any way will work. :+1:

### CODE REVIEW

**Comment by zeroSteiner** [comment]
This breaks Python 2.7 compatibility because the `ipaddress` module was added in [v3.3](https://docs.python.org/3/library/ipaddress.html). :cry: I don't know if I can add this without dropping that or some kinda of work around.

**Comment by Script-Nomad** [comment]
Super simple fix. Sorry i didn't get to the change request earlier. Had a very busy month.

**Comment by Script-Nomad** [comment]
While I'm in a coding mood, I included a function that can also sort IP addresses according to their numerical sort-order. This method also supports port numbers a-la `1.2.3.4:5` format

**Comment by zeroSteiner** [comment]
image

File: smoke_zephyr/utilities.py [hunk 1]
```diff
@@ -530,6 +532,52 @@ def is_valid_email_address(email_address):
...
+		if mask and '/' not in ip_network:
+			net = netaddr.IPNetwork("{0}/{1}".format(ip_network, mask))
+			return [host.__str__() for host in net.iter_hosts()]
+		elif '/' not in ip_network:
+			return [str(netaddr.IPAddress(ip_network))]
+		else:
+			net = netaddr.IPNetwork(ip_network)
+			return [host.__str__() for host in net.iter_hosts()]
+
+def sort_ip_list(ip_list, unique=True):
```
Comment (zeroSteiner) [inline smoke_zephyr/utilities.py:563]: I'm going to rename this to `sort_ipv4_list` to be a bit more explicit until we have IPv6 support (if we ever have IPv6 support). You make it really clear in the docs, but I think renaming the function would be helpful as well.

File: smoke_zephyr/utilities.py [hunk 2]
```diff
@@ -530,6 +532,52 @@ def is_valid_email_address(email_address):
 	# requirements = re
 	return EMAIL_REGEX.match(email_address) != None
 
+def get_ip_list(ip_network, mask=None):
+	"""
+	Quickly convert an IPv4 or IPv6 network (CIDR or Subnet) to a list
+	of individual IPs in their string representation.
+
+	:param str ip_network:
+	:param str mask:
```
Comment (zeroSteiner) [inline smoke_zephyr/utilities.py:541]: It looks like mask should actually be an `int` between 0 and 32 inclusive.

**Review by zeroSteiner — APPROVED** [review]
Left two minor comments. I'll give this a quick test and if things go smoothly make the minor changes myself. Thanks!

**Comment by zeroSteiner** [comment]
Well I spoke too soon, the `netaddr` module is not in the standard library so the tests are still broken. After thinking about it some more, I think it's time to drop Python 2.7 support anyways. The last release on PyPi will still be present and available for projects depending on Smoke Zephyr but there's not much point in bending over backwards to support 2.7 when it's EoL'ed anyways.

So having said that, I'm going to proceed with the Python 3 version of this, bump the major version to imply a backwards incompatible change and roll with it.

**Comment by zeroSteiner** [comment]
Merged, thanks @True-Demon! I made the following changes in commit e8a1cc7b7e0b803ebd0d53a85982845fd5953e5b.

* Added the functions to the docs
* Bumped the version to 2.0.0
* Added some basic unit testing
* Made the minor changes I left from my review.

**Comment by zeroSteiner** [comment]
FYI, version 2.0 has now been released and is up on PyPi including these changes. Support for versions 2.7 and 3.3 was dropped.

**Pull request merged**